### ISSUE TITLE
System.ArgumentException: Version string portion was too short or too long. 

### ISSUE THREAD

**Issue by ZeWaka** [tgstation/IconDiffBot#72]
```
System.ArgumentException: Version string portion was too short or too long. (Parameter 'input')
   at System.Version.ParseVersion(ReadOnlySpan`1 input, Boolean throwOnFailure)
   at System.Version.Parse(String input)
   at IconDiffBot.Core.DiffGenerator.BuildDmi(String metadata)
   at IconDiffBot.Core.DiffGenerator.GenerateDiffs(Stream before, Stream after)
   at IconDiffBot.Core.PayloadProcessor.<>c__DisplayClass12_0.<<GenerateDiffs>g__DiffDmi|0>d.MoveNext()
```
a) The commit that caused the failure
https://github.com/goonstation/goonstation/pull/7226
https://github.com/Skeletonman0/goonstation/commit/45accdd0758a449cc808f336b3e2a5d63281a27d
https://github.com/Skeletonman0/goonstation/blob/0cb66d9e0cd362feac9a5bfbf495446045057d0c/icons/turf/walls_ancient.dmi
https://github.com/Skeletonman0/goonstation/blob/0cb66d9e0cd362feac9a5bfbf495446045057d0c/icons/turf/walls_cave.dmi

b) The failed check run
https://github.com/goonstation/goonstation/pull/7226/checks?check_run_id=4876785039

**Comment by ZeWaka** [comment]
Still reproduces in a re-ran check, so it's not completely transient: https://github.com/goonstation/goonstation/pull/7226/checks?check_run_id=4877898512

**Comment by SpaceManiac** [comment]
walls_ancient.dmi and walls_cave.dmi (walls_overgrown is fine) as of that commit has metadata that begins like:
```
# BEGIN DMI
version = 
	width = 32
	height = 32
```
when it should look like:
```
# BEGIN DMI
version = 4.0
	width = 32
	height = 32
```
Probably IconDiffBot should not crash here, and anyways should not be using `System.Version` to represent this field, which is _not_ "the version number of an assembly, operating system, or the common language runtime".

**Comment by ZeWaka** [comment]
Interesting, those were both created with Dream Maker, so I blame DM for generating a malformed file.
I'd definitely argue it shouldn't crash and default to 4, especially if the program is liable to omit such information.

**Comment by ArcaneMusic** [comment]
I've been running into the same issue on https://github.com/wall-nerds/wallening/pull/83 .
I saw that the /tg/ cutting tool that we were using for this (Which apparently also runs into the same issue with tables as it does with walls!) wasn't adding the version variable as an argument. However, by manually adding that version 4.0 to the file through tweakPNG, it now claims that all of those files are now out of date... AND still can't be read by icondiffbot! Except for the pizza walls, for some reason.
[icerock_wall.zip](https://github.com/tgstation/IconDiffBot/files/9355413/icerock_wall.zip)
 I don't know how much of it is my end and how much of it is from trying to force a solution, but I found this issue now after working on this for about an hour.

**Comment by Imaginos16** [comment]
> walls_ancient.dmi and walls_cave.dmi (walls_overgrown is fine) as of that commit has metadata that begins like:
> 
> ```
> # BEGIN DMI
> version = 
> 	width = 32
> 	height = 32
> ```
> 
> when it should look like:
> 
> ```
> # BEGIN DMI
> version = 4.0
> 	width = 32
> 	height = 32
> ```
> 
> Probably IconDiffBot should not crash here, and anyways should not be using `System.Version` to represent this field, which is _not_ "the version number of an assembly, operating system, or the common language runtime".

Ran into an extremely similar issue when doing PR [#69147](https://github.com/tgstation/tgstation/pull/69147/checks?check_run_id=7796005707), but tweakpng seems to have fixed it. Unfortunately for Arcane's case, doing so did not fix his.

**Comment by ZeWaka** [comment]
Fixed by tgstation/icon-cutter#11
